tests/templates/test_survey_site_visit_data_v2.py

```diff
@@ -112,13 +126,19 @@ def mocked_schema(self, mocker: pytest_mock.MockerFixture) -> unittest.mock.Magi
         descriptor["fields"] = [f for f in descriptor["fields"] if f["name"] in fieldnames]
```

Comment:
This line is probably the source of schema not going back to normal.
It is mutating the `descriptor` dict which was returned by `schema()` and is stored in the cache.

Reply:
i thought that too and attempted to use `copy.deepcopy` to isolate it. Didn't work.